internal/component/beyla/ebpf/beyla_linux.go

```diff
@@ -116,43 +126,76 @@ func (args Services) Convert() (services.DefinitionCriteria, error) {
 }
 
 func (args KubernetesService) Convert() (map[string]*services.RegexpAttr, error) {
-	namespace, err := stringToRegexpAttr(args.Namespace)
-	if err != nil {
-		return nil, err
+	metadata := map[string]*services.RegexpAttr{}
+	if args.Namespace != "" {
+		namespace, err := stringToRegexpAttr(args.Namespace)
+		metadata[services.AttrNamespace] = &namespace
+		if err != nil {
+			return nil, err
+		}
 	}
-	podName, err := stringToRegexpAttr(args.PodName)
-	if err != nil {
-		return nil, err
+	if args.PodName != "" {
+		podName, err := stringToRegexpAttr(args.PodName)
+		metadata[services.AttrPodName] = &podName
+		if err != nil {
+			return nil, err
+		}
 	}
-	deploymentName, err := stringToRegexpAttr(args.DeploymentName)
-	if err != nil {
-		return nil, err
+	if args.DeploymentName != "" {
+		deploymentName, err := stringToRegexpAttr(args.DeploymentName)
+		metadata[services.AttrDeploymentName] = &deploymentName
+		if err != nil {
+			return nil, err
+		}
 	}
-	replicaSetName, err := stringToRegexpAttr(args.ReplicaSetName)
-	if err != nil {
-		return nil, err
+	if args.ReplicaSetName != "" {
+		replicaSetName, err := stringToRegexpAttr(args.ReplicaSetName)
+		metadata[services.AttrReplicaSetName] = &replicaSetName
+		if err != nil {
+			return nil, err
+		}
 	}
-	statefulSetName, err := stringToRegexpAttr(args.StatefulSetName)
-	if err != nil {
-		return nil, err
+	if args.StatefulSetName != "" {
+		statefulSetName, err := stringToRegexpAttr(args.StatefulSetName)
+		metadata[services.AttrStatefulSetName] = &statefulSetName
+		if err != nil {
+			return nil, err
+		}
 	}
-	dataaemonSetName, err := stringToRegexpAttr(args.DaemonSetName)
-	if err != nil {
-		return nil, err
+	if args.DaemonSetName != "" {
+		daemonSetName, err := stringToRegexpAttr(args.DaemonSetName)
+		metadata[services.AttrDaemonSetName] = &daemonSetName
+		if err != nil {
+			return nil, err
+		}
 	}
-	ownerName, err := stringToRegexpAttr(args.OwnerName)
-	if err != nil {
-		return nil, err
+	if args.OwnerName != "" {
+		ownerName, err := stringToRegexpAttr(args.OwnerName)
+		metadata[services.AttrOwnerName] = &ownerName
+		if err != nil {
+			return nil, err
+		}
 	}
-	return map[string]*services.RegexpAttr{
-		services.AttrNamespace:       &namespace,
-		services.AttrPodName:         &podName,
-		services.AttrDeploymentName:  &deploymentName,
-		services.AttrReplicaSetName:  &replicaSetName,
-		services.AttrStatefulSetName: &statefulSetName,
-		services.AttrDaemonSetName:   &dataaemonSetName,
-		services.AttrOwnerName:       &ownerName,
-	}, nil
+	return metadata, nil
+}
+
+func (args Prometheus) Convert() prom.PrometheusConfig {
+	p := beyla.DefaultConfig.Prometheus
+	if args.Features != nil {
+		p.Features = args.Features
+	}
+	if args.Instrumentations != nil {
+		p.Instrumentations = args.Instrumentations
+	}
+	return p
+}
+
+func (args Network) Convert() beyla.NetworkConfig {
+	networks := beyla.DefaultConfig.NetworkFlows
+	if args.Enable {
+		networks.Enable = true
+	}
```

Comment:
I'm not a big fan of the defaults being set in the convert functions. I think this is error-prone. For example, here it works because `enable` is by default "false" but if one day it is changed to true in Beyla, then the logic above would need to be updated because there would not be a way for the user to set it to false. The pattern used for all other components is to implement the "[SetToDefault()](https://github.com/grafana/alloy/blob/main/syntax/internal/value/decode.go#L30)" from the Defaulter interface. Is there a particular reason why it is not used for this component?

Reply:
This might be fine for this PR because of the short window that we have with Alloy's release cut but it would be nice to address this is a follow up PR